tools/testoperator/src/commands/evals/mod.rs

```diff
@@ -0,0 +1,171 @@
+        .body(body)
+        .send()
+        .await?;
+
+    if !response.status().is_success() {
+        return Err(anyhow::anyhow!(
+            "Failed to execute evals: {}",
+            response.text().await?
+        ));
+    }
+
+    println!("Execution completed, retrieving results...");
+
+    let mut flight_client = spiced_instance.flight_client().await?;
+
+    let eval_result = execute_sql(&mut flight_client, QUERY_EVAL_BENCHMARK_MAIN_METRICS).await?;
+    println!("Result:\n{}\n", pretty_format_batches(&eval_result)?);
+
+    let tasks_calls = execute_sql(&mut flight_client, QUERY_EVAL_BENCHMARK_TASKS).await?;
+    println!(
+        "Executed tasks:\n{}\n",
+        pretty_format_batches(&tasks_calls)?
+    );
+
+    let top_errors = execute_sql(&mut flight_client, QUERY_EVAL_BENCHMARK_TOP_ERRORS).await?;
+    println!("Top errors:\n{}\n", pretty_format_batches(&top_errors)?);
+
+    spiced_instance.stop()?;
+
+    println!("Benchmark completed");
+
+    Ok(())
+}
+
+async fn execute_sql(
+    flight_client: &mut FlightClient,
+    sql: &str,
+) -> Result<Vec<RecordBatch>, anyhow::Error> {
+    let res = flight_client
+        .query(sql)
+        .await?
+        .try_collect::<Vec<RecordBatch>>()
+        .await?;
+    Ok(res)
+}
+
+static QUERY_EVAL_BENCHMARK_MAIN_METRICS: &str = "
+WITH latest_run AS (
+    SELECT id, created_at, EXTRACT(EPOCH FROM (completed_at - created_at)) AS duration_seconds
+    FROM spice.eval.runs
+    ORDER BY created_at DESC LIMIT 1
+),
+score AS (
+    SELECT run_id, AVG(value) AS overall_score, COUNT(*) AS evals_count
+    FROM spice.eval.results
+    WHERE run_id = (SELECT id FROM latest_run)
+    GROUP BY run_id
+),
+tool_stats AS (
+    SELECT 
+        COUNT(*) AS task_calls,
+        COUNT(CASE WHEN error_message IS NOT NULL THEN 1 END) AS task_errors
+    FROM runtime.task_history
+    WHERE 
+        task != 'test_connectivity'
+        AND start_time BETWEEN (SELECT created_at FROM latest_run)
+        AND COALESCE(end_time, NOW())
+)
+SELECT r.id AS run_id, r.model, r.status, s.evals_count AS tests, lr.duration_seconds, ROUND(s.overall_score, 4) as score, ts.task_calls, ts.task_errors
+FROM spice.eval.runs r
+JOIN latest_run lr ON r.id = lr.id
+LEFT JOIN score s ON r.id = s.run_id
+LEFT JOIN tool_stats ts ON 1 = 1;
+";
+
+static QUERY_EVAL_BENCHMARK_TASKS: &str = "
+WITH latest_run AS (
+  SELECT id 
+  FROM spice.eval.runs 
+  ORDER BY created_at DESC 
+  LIMIT 1
+)
+SELECT 
+  task, 
+  COUNT(*) AS calls,
+  COUNT(CASE WHEN error_message IS NOT NULL THEN 1 END) AS failures,
+  SUM(CAST((end_time - start_time) AS Float) /  1000000) AS duration_ms
+FROM runtime.task_history
+WHERE 
+  task != 'test_connectivity'
+  AND start_time BETWEEN (SELECT created_at FROM spice.eval.runs WHERE id = (SELECT id FROM latest_run)) AND 
+  COALESCE(end_time, NOW())
+GROUP BY task
+ORDER BY duration_ms DESC;
+";
+
+static QUERY_EVAL_BENCHMARK_TOP_ERRORS: &str = "
```

Comment:
Same for all these

Reply:
Documented, added example output - thank you!